src/transaction.js

```diff
@@ -201,11 +192,13 @@ Transaction.prototype.hashForSignature = function (inIndex, prevOutScript, hashT
 
   // in case concatenating two scripts ends up with two codeseparators,
   // or an extra one at the end, this prevents all those possible incompatibilities.
-  var hashScript = prevOutScript.without(opcodes.OP_CODESEPARATOR)
+  var hashScript = scripts.compile(scripts.decompile(prevOutScript).filter(function (x) {
```

Comment:
As a general algorithm,  `without` did not handle data chunks correctly.
Now that we can use `filter` by itself,  the semantics are clear and it is on the user to know that `Buffer`s are not comparable without using a `.compare` or a `bufferutils.equals` equivalent.

In that sense, there is no need for a `TODO` here, we only care about the `opcodes.OP_CODESEPARATOR` byte *not* being the result.